community/cypher/acceptance/src/test/scala/org/neo4j/internal/cypher/acceptance/AggregationAcceptanceTest.scala

```diff
@@ -243,4 +243,19 @@ class AggregationAcceptanceTest extends ExecutionEngineFunSuite with NewPlannerT
 
     result.toList should equal(List(Map("foo" -> 42, "bar" -> 42, "baz" -> Map("y" -> 1))))
   }
+
+  test("should not project too much when aggregating in a WITH before merge and after a WITH with filter") {
```

Comment:
I don't think you can trigger it from the outside like that - the semantic check will fail long before the bug would be triggered.